src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -1906,12 +1906,39 @@ private IEnumerable<PENamedTypeSymbol> CreateNestedTypes()
             {
                 if (this.SpecialType == Microsoft.CodeAnalysis.SpecialType.None)
                 {
-                    isOrdinaryStruct = true;
                     isOrdinaryEmbeddableStruct = this.ContainingAssembly.IsLinked;
                 }
-                else
-                {
-                    isOrdinaryStruct = (this.SpecialType == Microsoft.CodeAnalysis.SpecialType.System_Nullable_T);
```

Comment:
>(this.SpecialType == Microsoft.CodeAnalysis.SpecialType.System_Nullable_T) [](http://example.com/codeflow?start=39&length=74)

It looks like there is a similar logic on VB side. We need to apply similar change, and, since we don't have a failing test for that code, add a unit-tests that is failing in commit 60.

Reply:
Found corresponding VB code, but it seems that it is very poorly tested. E.g. replacing [this check](https://github.com/dotnet/roslyn/blob/b701a39ca8ccec581a7949d3bcc9931a2088deee/src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.vb#L1141) with just `If True Then` doesn't make _any_ test fail in `MS.CA.VB.Symbols.UnitTests` project. On C# side there was a failing test of `UnmanagedCallersOnly` attribute, but I don't think it is a VB feature. So, could you please point out what kind of test we can add to verify such ‘ordinary struct’ behavior for VB?

Reply:
> Found corresponding VB code, but it seems that it is very poorly tested. 

I would say C# is not in a better state. That code is probably 13 years old, if not more, but the ```UnmanagedCallersOnly``` is fairly recent feature. Also, it is hard to add a targeted regression test for the logic, because it is almost impossible to predict what type will be added to the `SpecialType` set next. In cases like this, using a future-proof coding techniques is the best option. One of the techniques is to check for what you are looking for, rather than checking for something you are not looking for. I.e. check for types you want to treat specially, rather than checking for types that you don't want to treat specially. 

> So, could you please point out what kind of test we can add to verify such ‘ordinary struct’ behavior for VB?

Behavior of that code is observable through an API, we can check what fields were imported for the `ReadOnlySpan` type defined in a core library. Consider adding similar test for C# as well. The recommended location for the tests:
- src\Compilers\VisualBasic\Test\Symbol\SymbolsTests\Metadata\PE\LoadingFields.vb
- src\Compilers\CSharp\Test\Symbol\Symbols\Metadata\PE\LoadingFields.cs